flocker/node/agents/ebs.py

```diff
@@ -21,32 +23,49 @@
 from boto.exception import EC2ResponseError
 from twisted.python.filepath import FilePath
 
+from eliot import Message
+
 from .blockdevice import (
     IBlockDeviceAPI, BlockDeviceVolume, UnknownVolume, AlreadyAttachedVolume,
     UnattachedVolume, get_blockdevice_volume,
 )
 from ._logging import (
     AWS_ACTION, BOTO_EC2RESPONSE_ERROR, NO_AVAILABLE_DEVICE,
-    NO_NEW_DEVICE_IN_OS, WAITING_FOR_VOLUME_STATUS_CHANGE
+    NO_NEW_DEVICE_IN_OS, WAITING_FOR_VOLUME_STATUS_CHANGE,
+    BOTO_LOG_HEADER,
 )
 
 DATASET_ID_LABEL = u'flocker-dataset-id'
 METADATA_VERSION_LABEL = u'flocker-metadata-version'
 CLUSTER_ID_LABEL = u'flocker-cluster-id'
 ATTACHED_DEVICE_LABEL = u'attached-device-name'
-BOTO_NUM_RETRIES = u'10'
+BOTO_NUM_RETRIES = u'20'
 VOLUME_STATE_CHANGE_TIMEOUT = 300
 
-# Set Boto debug level to ``1``, requesting basic debug messages from Boto
-# to be printed.
-# See https://code.google.com/p/boto/wiki/BotoConfig#Boto
-# for available debug levels.
-# Warning: BotoConfig is a Boto global option, so, all Boto clients from
-# from this process will have their log levels modified. See
-# https://clusterhq.atlassian.net/browse/FLOC-1962
-# to track evaluation of impact of log level change.
-BOTO_DEBUG_LEVEL = u'1'
+class EliotLogHandler(logging.Handler):
+    _to_log = {"Method", "Path", "Params"}
+    def emit(self, record):
+        fields = vars(record)
+        # Only log certain things.  The log is massively too verbose
+        # otherwise.
+        if fields.get("msg", ":").split(":")[0] in self._to_log:
+            Message.new(
+                message_type=BOTO_LOG_HEADER, **fields
+            ).write()
+
+def _enable_boto_logging():
+    """
+    Make boto log activity using Eliot.
+    """
+    logger = logging.getLogger("boto")
+    logger.addHandler(EliotLogHandler())
 
+    # It seems as though basically all boto log messages are at the same
+    # level.  Either we can see all of them or we can see none of them.
+    # We'll do some extra filtering in the handler.
+    logger.setLevel(logging.DEBUG)
```

Comment:
When i set boto log level to ``logging.DEBUG``, the amount of information output was way too much and not so useful for debuggability. So, reverted to ``logging.ERROR``.